exchanges/graphcache/src/operations/write.ts

```diff
@@ -246,9 +255,14 @@ const writeSelection = (
       }
     }
 
-    if (fieldName === '__typename') {
-      continue;
-    } else if (ctx.optimistic && isRoot) {
+    // We simply skip all typenames fields and assume they've already been written above
+    if (fieldName === '__typename') continue;
+
+    // Add the current alias to the walked path before processing the field's value
+    ctx.__internal.path.push(fieldAlias);
```

Comment:
I feel like the pushing / popping to `path` is quite verbose. But it seems fine and I can't think of a better way of doing it off the top of my head

Reply:
Yea, this was the simplest implementation I could come up with. I tried to merge it into the `SelectionIterator` stuff, but then it wouldn't capture the indices, so this is indeed the simplest way to get that path stack in there